langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AbstractAzureAiSearchEmbeddingStore.java

```diff
@@ -263,6 +264,48 @@ public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedde
         return ids;
     }
 
+    @Override
+    public void remove(String id) {
+        ensureNotBlank(id, "id");
+        this.removeAll(singletonList(id));
+    }
+
+    @Override
+    public void removeAll(Collection<String> ids){
+        ensureNotEmpty(ids, "ids");
+        List<Map<String, String>> documentsToDelete = new ArrayList<>();
+        for (String id : ids) {
+            ensureNotBlank(id, "id");
+            Map<String, String> documents = new HashMap<>();
+            documents.put(DEFAULT_FIELD_ID, id);
+            documentsToDelete.add(documents);
+        }
+        searchClient.deleteDocuments(documentsToDelete);
```

Comment:
Should we inspect the result of the operation?
For example:
```
        var indexDocumentsResult = searchClient.deleteDocuments(documentsToDelete);
        var errorMessage = indexDocumentsResult.getResults().stream()
                .filter(not(IndexingResult::isSucceeded))
                .reduce("", (acc, r) -> acc +
                                "Document %s could not be deleted. Status: %s , error: %s\n".formatted(r.getKey(), r.getStatusCode(), r.getErrorMessage()),
                        String::concat);
        if (!errorMessage.isEmpty()) {
            throw new AzureException(errorMessage);
        }
```

Reply:
good point!